src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -1806,7 +1805,9 @@ public void onNodeStarted(DiscoveryNode localNode) {
         if (!SSLConfig.isSslOnlyMode() && !client && !disabled) {
             cr.initOnNodeStart();
         }
-        this.localNode = localNode;
+        if (!SSLConfig.isSslOnlyMode() && si != null) {
+            si.setLocalNode(localNode); // set localNode
+        }
```

Comment:
Followed same logic as on line 917 where si [is initialized](https://github.com/opensearch-project/security/blob/f6f2444223721ff3955c8faa720f5e43263a4b93/src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java#L1052) only if the if condition below is false: https://github.com/opensearch-project/security/blob/f6f2444223721ff3955c8faa720f5e43263a4b93/src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java#L917-L931

Reply:
Switching to AtomicReference fixed this issue, IMO much cleaner, I've pushed this change.